src/gatherFiles.js

```diff
@@ -0,0 +1,53 @@
+import { basename, join } from 'path';
+import { cwd } from 'process';
+import glob from 'glob';
+import { statSync } from 'fs';
+
+function getFilesRecursively(path, {
+  globString = '**/*.{js,jsx,js.map}',
+  map,
+  realPaths,
+  workingDirectory = cwd(),
+} = {}) {
+  const fullPath = join(workingDirectory, path);
+  if (statSync(fullPath).isFile()) {
+    if (!realPaths.has(fullPath)) {
+      map.push({
+        path: fullPath,
+        name: basename(fullPath),
+      });
+      realPaths.add(fullPath);
+    }
+  } else {
+    try {
+      glob(globString, { cwd: fullPath, sync: true }).forEach(file =>
+        getFilesRecursively(file, {
+          globString,
+          map,
+          realPaths,
+          workingDirectory: fullPath,
+        })
+      );
+    } catch (e) {} // eslint-disable-line no-empty
```

Comment:
Not sure exactly -- other than I don't actually know of any errors that might occur (no mention of them in the [`glob` library](https://www.npmjs.com/package/glob) documentation) -- our existing upload command doesn't handle any potential glob errors, so I kind of figured it would be okay to continue in that same way.